flytekit/common/tasks/sdk_runnable.py

```diff
@@ -400,19 +400,30 @@ def execute(self, context, inputs):
                     literals={k: v.sdk_value for k, v in _six.iteritems(outputs_dict)}
                 )
             }
+        # New style - V1: inputs are just inputs
         else:
-            outputs = self._execute_user_code(context, inputs_dict)
-            sdk_types = {k: _type_helpers.get_sdk_type_from_literal_type(v.type) for k, v in self.interface.outputs.items()}
+            new_style_inputs = flyte_engine.idl_literal_map_to_python_value(context, inputs)
+            outputs = self._execute_user_code(context, new_style_inputs)
             expected_output_names = list(self.interface.outputs.keys())
             if len(expected_output_names) == 1:
                 literals = {expected_output_names[0]: outputs}
             else:
-                literals = {expected_output_names[i]: outputs[i] for i, _ in enumerate(outputs)} 
+                # Question: How do you know you're going to enumerate them in the correct order? Even if autonamed, will
```

Comment:
ah because you are saying there is no expectation from proto representation of the interface about ordering to begin with?
{"a": int, "b": str} is not the same as {"b": str, "a": int} from proto perspective I agree... however, in go, for example, we don't get to choose in which ordering does it get loaded into a go map... so if we ever build a golang SDK, we won't be able to apply that same logic...

Maybe the option to reinspect the annotations to match up the names is not a bad one... I was hoping we don't need to do that during execution though... I was hoping we could, one day, do something like:
```
run(my_mock_python_func, idl_task_template)
```
but maybe that's a bit of wishful thinking... we can still achieve the same by requiring the mock func to adhere to the same interface with the same names... that's also ok....

Alright, sorry for writing down my thoughts as they flow :) I think I like re-inspecting the func python interface... just not right here... probably somewhere a bit higher when we load the python module & func and match it up to the idl interface before proceeding...